### PR TITLE
[Intermediate Representation]  xgboost predict using IR

### LINKED ISSUE #1051: [Intermediate Representation] XGBoost predict using IR
Transfer the xgboost predict logic to PredictIR

### PR DESCRIPTION
**Pull request by Yancey0623**
fixed #1051

### CODE REVIEW

File: pkg/sql/codegen/xgboost/template_pred.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+feature_field_meta = json.loads('''{{.FieldMetaJSON}}''')
+label_field_meta = json.loads('''{{.LabelJSON}}''')
+
+feature_column_name = sorted([k["name"] for k in feature_field_meta])
+label_name = label_field_meta["name"]
+
+feature_spec = {k['name']: k for k in feature_field_meta}
+
+conn = connect_with_data_source('''{{.DataSource}}''')
```
Comment (weiguoz) [inline pkg/sql/codegen/xgboost/template_pred.go:42]: from sqlflow_submitter.db import connect_with_data_source

File: pkg/sql/executor.go [hunk 2]
```diff
@@ -497,6 +498,13 @@ func loadModelMeta(pr *extendedSelect, db *DB, cwd, modelDir, modelName string)
 	return pr, fts, nil
 }
 
+func enableIR() bool {
+	if os.Getenv("SQLFLOW_codegen") == "ir" {
+		return true
+	}
+	return false
```
Comment (weiguoz) [inline pkg/sql/executor.go]: `return os.Getenv("SQLFLOW_codegen") == "ir"`

File: pkg/sql/ir_generator.go [hunk 3]
```diff
@@ -98,10 +98,15 @@ func generatePredictIR(slct *extendedSelect, connStr string, cwd string, modelDi
 	}
 	fmt.Printf("select %s, result table: %s\n", slct.standardSelect.String(), slct.into)
```
Comment (weiguoz) [inline pkg/sql/ir_generator.go]: Remove `fmt.Printf...`

File: pkg/sql/ir_generator.go [hunk 4]
```diff
@@ -558,3 +563,16 @@ func parseShape(e *expr) ([]int, error) {
 	}
 	return shape, nil
 }
+
+func parseResultTable(intoStatement string) (string, error) {
+	resultTableParts := strings.Split(intoStatement, ".")
+	resultTable := ""
+	if len(resultTableParts) == 3 {
```
Comment (weiguoz) [inline pkg/sql/ir_generator.go:573]: I'm curious the value of `len(resultTableParts)` would be 3, 2 and 1...
Would you plz add some comments? Thanks.

File: pkg/sql/ir_generator.go [hunk 5]
```diff
@@ -558,3 +563,16 @@ func parseShape(e *expr) ([]int, error) {
...
+		resultTable = strings.Join(resultTableParts[0:2], ".")
+	} else if len(resultTableParts) == 2 || len(resultTableParts) == 1 {
+		resultTable = intoStatement
+	} else {
+		return "", fmt.Errorf("error result table format, should be db.table.class_col or db.table or table")
+	}
+	return resultTable, nil
```
Comment (weiguoz) [inline pkg/sql/ir_generator.go]: How about making the condition more clearly?
``` go
    resultTableParts := strings.Split(intoStatement, ".")
    if len(resultTableParts) > 3 {
        return "", fmt.Errorf("bla")
    } else if len (resultTableParts) == 3 {
        return strings.Join(resultTableParts[0:2], "."), nill
    } else {
        return intoStatement, nil
    }
```

Never mind, just for your reference.

Comment (tonyyang-svail) [reply pkg/sql/ir_generator.go]: FYI, package `tensorflow` has the same code snippet. If we were to change them, change both.
https://github.com/sql-machine-learning/sqlflow/blob/0b716a3b1bdcd038ddb401fe292a9169c6560e41/pkg/sql/codegen/tensorflow/codegen.go#L241-L249

File: pkg/sql/executor.go [hunk 6]
```diff
@@ -28,6 +28,7 @@ import (
 
 	pb "sqlflow.org/sqlflow/pkg/server/proto"
 	"sqlflow.org/sqlflow/pkg/sql/codegen/tensorflow"
+	"sqlflow.org/sqlflow/pkg/sql/codegen/xgboost"
```
Comment (tonyyang-svail) [inline pkg/sql/executor.go:31]:
```
"sqlflow.org/sqlflow/pkg/sql/codegen/xgboost"
xgb "sqlflow.org/sqlflow/pkg/sql/codegen/xgboost"
```
Looks like the package has been imported twice. Not sure if there is any side effect.

**Review by tonyyang-svail — DISMISSED** [review]
Generally LGMT. Please also refer to the comments by @weiguoz and me for possible improvements.

**Review by weiguoz — APPROVED** [review]
LGTM

**Pull request merged**